src/ai/backend/web/proxy.py

```diff
@@ -127,6 +148,12 @@ async def web_handler(request, *, is_anonymous=False) -> web.StreamResponse:
             # but need to keep the client's version header so that
             # the final clients may perform its own API versioning support.
             request_api_version = request.headers.get('X-BackendAI-Version', None)
+            secure_context = request.headers.get('X-BackendAI-Encoded', None)
```

Comment:
I think the naming "Encoded" is misleading because encryption != encoding. How about "X-BackendAI-Encrypted"?